manifests/init.pp

```diff
@@ -110,64 +102,117 @@
+  Boolean $enable_docker = true,
+  Boolean $enable_deb = true,
+
+  Stdlib::Absolutepath $repo_export_dir = '/var/lib/pulp/katello-export',
+
+  String $candlepin_db_host = 'localhost',
+  Optional[Stdlib::Port] $candlepin_db_port = undef,
+  String $candlepin_db_name = 'candlepin',
+  String $candlepin_db_user = 'candlepin',
+  Optional[String] $candlepin_db_password = undef,
+  Boolean $candlepin_db_ssl = false,
+  Boolean $candlepin_db_ssl_verify = true,
+  Boolean $candlepin_manage_db = true,
 
-  String $candlepin_oauth_key = $katello::params::candlepin_oauth_key,
-  String $candlepin_oauth_secret = $katello::params::candlepin_oauth_secret,
+  String $pulp_db_name = 'pulp_database',
+  String $pulp_db_seeds = 'localhost:27017',
+  Optional[String] $pulp_db_username = undef,
+  Optional[String] $pulp_db_password = undef,
+  Optional[String] $pulp_db_replica_set = undef,
+  Boolean $pulp_db_ssl = false,
+  Optional[Stdlib::Absolutepath] $pulp_db_ssl_keyfile = undef,
+  Optional[Stdlib::Absolutepath] $pulp_db_ssl_certfile = undef,
+  Boolean $pulp_db_verify_ssl = true,
+  Stdlib::Absolutepath $pulp_db_ca_path = '/etc/pki/tls/certs/ca-bundle.crt',
+  Boolean $pulp_db_unsafe_autoretry = false,
+  Optional[Enum['majority', 'all']] $pulp_db_write_concern = undef,
+  Boolean $pulp_manage_db = true,
+) {
 
-  Integer[0] $rest_client_timeout = $katello::params::rest_client_timeout,
-  Integer[0, 1000] $qpid_wcache_page_size = $katello::params::qpid_wcache_page_size,
-  String $qpid_interface = $katello::params::qpid_interface,
-  String $qpid_hostname = $katello::params::qpid_hostname,
-  Integer[1] $num_pulp_workers = $katello::params::num_pulp_workers,
-  Integer[0] $pulp_worker_timeout = $katello::params::pulp_worker_timeout,
-  Optional[Stdlib::HTTPUrl] $proxy_url = $katello::params::proxy_url,
-  Optional[Integer[0, 65535]] $proxy_port = $katello::params::proxy_port,
-  Optional[String] $proxy_username = $katello::params::proxy_username,
-  Optional[String] $proxy_password = $katello::params::proxy_password,
-  Optional[String] $pulp_max_speed = $katello::params::pulp_max_speed,
-  Optional[Enum['SSLv23', 'TLSv1']] $cdn_ssl_version = $katello::params::cdn_ssl_version,
+  package { 'katello':
+    ensure => installed,
+  }
 
-  Array[String] $package_names = $katello::params::package_names,
-  Boolean $enable_ostree = $katello::params::enable_ostree,
-  Boolean $enable_yum = $katello::params::enable_yum,
-  Boolean $enable_file = $katello::params::enable_file,
-  Boolean $enable_puppet = $katello::params::enable_puppet,
-  Boolean $enable_docker = $katello::params::enable_docker,
-  Boolean $enable_deb = $katello::params::enable_deb,
+  class { 'katello::globals':
+    enable_ostree => $enable_ostree,
+    enable_yum    => $enable_yum,
+    enable_file   => $enable_file,
+    enable_puppet => $enable_puppet,
+    enable_docker => $enable_docker,
+    enable_deb    => $enable_deb,
+  }
 
+  class { 'katello::params':
```

Comment:
Never seen calling params and configuring it as a design convention.

Reply:
It's unconventional and I might clean this up a bit more later as I gain more experience with splitting up into cleaner roles and profiles.